### PR TITLE
Feature/profile marketer

### PR DESCRIPTION
**Pull request by lili2311**
- Adding in more profile fields
- Adding in a new form to change profile image on profile page

Please review only

### CODE REVIEW

File: pimpmycause/pimpmycause/templates/profiles/profile.html [hunk 1]
```diff
@@ -6,39 +6,93 @@
...
-                                    </div>                                
+                            </div>
+                            <div class="row">
+                                <h2> Social</h2>
+                            </div>
+                            <div class="row">
+                                <div class="large-6 medium-6 small-12 columns">
+                                    {{ profile_update_form.linkedin.errors }}
+                                    <label for="linkedin"> LinkedIn </label>
+                                    <input type="url" name="linkedin" placeholder="LinkedIn" value='{{ profile_update_form.linkedin.value|default_if_none:"" }}' required>
```
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: if the field is required do we need like a red '*' and an explanation saying it's required ? or something else that helps the user understand it's required ? :)

File: pimpmycause/pimpmycause/templates/profiles/profile.html [hunk 2]
```diff
@@ -6,71 +6,93 @@
...
+
+                                    {{ profile_update_form.twitter.errors }}
+                                    <label for="twitter"> Twitter </label>
+                                    <input type="url" name="twitter" placeholder="Twitter" value='{{ profile_update_form.twitter.value|default_if_none:"" }}' required>
+
+                                    {{ profile_update_form.website.errors }}
+                                    <label for="website"> Website </label>
+                                    <input type="url" name="website" placeholder="www.example.com" value='{{profile_update_form.website.value|default_if_none:""}}' required>
```
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: it seems like it wants http:// or https:// before www to take it as a url ..

File: pimpmycause/pimpmycause/templates/profiles/profile.html [hunk 3]
```diff
@@ -6,39 +6,93 @@
...
+                                </div>
+                            </div>
+                            <div class="row">
+                                <h2>Experience</h2>
+                            </div>
+                            <div class="row">
+                                <div class="large-12 medium-12 small-12 columns">
+                                        {{ profile_update_form.bio.errors }}
+                                        <label for="bio"> Bio </label>
+                                        <textarea name="bio" rows="20" required placeholder="Write a short bio">
```
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: managed to update my profile without bio details - also couldn't see the placeholder (maybe i got that wrong so let me know please) - also the cursor doesn't start from the top left corner in the text area..
image

File: pimpmycause/pimpmycause/templates/profiles/profile.html [hunk 4]
```diff
@@ -6,39 +6,93 @@
...
+                                <div class="large-6 medium-6 small-12 columns">
+                                    <div class="large-6 medium-6 small-6 columns">
+                                        {{ profile_update_form.image.errors }}
+
+                                        {% if not profile_update_form.image.value %}
+                                            <img src="https://freeiconshop.com/files/edd/person-flat.png" alt="Profile Image">
+                                            <div class="small-12 columns">
+
+<!--                                           <label for="image"> Upload profile picture </label>
+ -->                                           <input class="button" type="submit" value="Upload Picture">
```
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: tried to change picture and didn't work :(

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 1 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: yep thanks haven't worked on that yet, will ad this in. We agreed with Anna we instead say when something is Optional

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 2 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: yes it is a url field so needs to be a real link

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 3 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: this one i spotted, i am trying to figure out this mystery I think Django is adding space when saving, perhaps need to trim the string will fix

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 4 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: For this one can you walk me through what you did so I can try to replicate? 
Did you choose a new file and hit update? note for now it won't show you a previous of new image that has not been implemented

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 2 above)
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: could we change the placeholder to say something like http://www.example.com ? or no need ?

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 3 above)
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: go!

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 4 above)
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: I clicked on the "Upload Picture" button and nothing came up..

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 1 above)
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: okies!

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 2 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: yep will change that will be nice and clear

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 4 above)
Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: I see I will take a look, did you check out the branch and install all the new requirements I assume?

Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: I think I know whats wrong, I commented out the bit that does stuff !

Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: Oops maybe my bad (didn't install the requirements)..just checked out the branch .will try again tonight

File: pimpmycause/pimpmycause/templates/profiles/profile.html (hunk 1 above)
Comment (menecio) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: You could profit of browsers `required` attribute support and just `<input name="something" required="true"/>`

Comment (lili2311) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: yep already using those, will recheck it matches actual required fields correctly, but django also validates this for you

File: pimpmycause/pimpmycause/templates/profiles/profile.html [hunk 5]
```diff
@@ -26,65 +30,54 @@ <h1 class="pnl__title">Profile</h1>
...
+                                <h2>Background</h2>
                             </div>
                             <div class="row">
                                 <div class="large-12 medium-12 small-12 columns">
                                         {{ profile_update_form.bio.errors }}
                                         <label for="bio"> Bio </label>
-                                        <textarea name="bio" rows="20" required placeholder="Write a short bio">
-                                        {{ profile_update_form.bio.value|default_if_none:"" }}
-                                        </textarea>
+                                        <textarea name="bio" rows="10" required placeholder="Hi! My name is Sam, I am a London based Marketer.">{{ profile_update_form.bio.value|default_if_none:"" }}</textarea>
```
Comment (yanabar) [inline pimpmycause/pimpmycause/templates/profiles/profile.html]: nice! you fixed it :)

File: pimpmycause/profiles/forms.py [hunk 6]
```diff
@@ -29,4 +29,19 @@ class Meta:
...
+    images = forms.URLField(widget=S3DirectWidget(
+        dest='user-profile-images',
+        html=(
+            '<div class="s3direct" data-policy-url="{policy_url}">'
+            '  <a class="file-link" target="_blank" href="{file_url}">{file_name}</a>'
+            '  <a class="file-remove" href="#remove">Remove</a>'
+            '  <input class="file-url" type="hidden" value="{file_url}" id="{element_id}" name="{name}" />'
+            '  <input class="file-dest" type="hidden" value="{dest}">'
+            '  <input class="file-input js-input-file inputfile" type="file" id="file" name="file"/>'
+            '  <label for="file" class="button">Choose an image</label>'
```
Comment (yanabar) [inline pimpmycause/profiles/forms.py]: There is an 'Images:' text just above the 'Choose an image' button which I cannot see in the code.. 
image

**Comment by yanabar** [comment]
tried it again - after resetting the variables locally 
image
 terminal now says:
image

**Comment by lili2311** [comment]
Okay thanks, I will try to push this to Heroku as a test and see if using one key from one place will solve this issue or alternatively grant you an access key

**Comment by yanabar** [comment]
👍

**Pull request merged**